modules/router/src/test/services/config.spec.ts

```diff
@@ -23,4 +45,70 @@ describe("config.ts", () => {
       expect(res.getValue()).to.be.deep.eq(config.rebalanceProfiles[0]);
     });
   });
+
+  describe("getMatchingSwap", () => {});
+
+  describe("getSwapFees", () => {});
+
+  describe("getMappedAssets", () => {
+    it("happy: getMappedAssets", () => {
+      const res = getMappedAssets(fromAssetId, fromChainId);
+      console.log(res);
+    });
+  });
+
+  describe("getPriceImpact", () => {
+    it("happy: getMappedAssets 1:1", () => {
+      const marketPrice = parseEther("1");
+      const estimatedPrice = parseEther("1");
+      const res = getPriceImpact(marketPrice, estimatedPrice);
+      console.log(res);
+    });
+
+    it("happy: getMappedAssets 1:1.2", () => {
+      const marketPrice = parseEther("1");
+      const estimatedPrice = parseEther("1.2");
+      const res = getPriceImpact(marketPrice, estimatedPrice);
+      console.log(res.toString());
+    });
+  });
+
+  describe("onSwapGivenIn", () => {
+    it("error if getOnchainBalance errors", async () => {
+      ethReader.getOnchainBalance.onFirstCall().resolves(Result.fail(new ChainError("getOnchainBalance error")));
+      const res = await onSwapGivenIn(
+        transferAmount,
+        fromAssetId,
+        fromChainId,
+        toAssetId,
+        toChainId,
+        routerSignerAddress,
+        ethReader as IVectorChainReader,
+        log,
+      );
+
+      expect(res.isError).to.be.true;
+      expect(res.getError()!.message).to.be.eq(ConfigServiceError.reasons.CouldNotGetAssetBalance);
+    });
+
+    it("error if provider isn't provided", async () => {
+      ethReader.getOnchainBalance.onFirstCall().resolves(Result.ok(parseEther("100")));
+      ethReader.getOnchainBalance.onSecondCall().resolves(Result.ok(parseEther("100")));
+      const res = await onSwapGivenIn(
+        transferAmount,
+        fromAssetId,
+        fromChainId,
+        toAssetId,
+        toChainId,
+        routerSignerAddress,
+        ethReader as IVectorChainReader,
+        log,
+      );
+
+      expect(res.isError).to.be.true;
+      expect(res.getError()!.message).to.be.eq(ConfigServiceError.reasons.UnableToGetSwapRate);
+    });
```

Comment:
Think we need one more test here to check for desired behavior in a valid use case. I will add shortly